en/middleware/mavlink.md

```diff
@@ -114,28 +121,83 @@ protected:
+  For your own streaming classes these can just be copied and modified to match the values for your MAVLink message.
+- The `private` definitions subscribe to the uORB topics that need to be published.
+  Here we also define constructors to prevent the definition being copied.
+- The `protected` section is where the important work takes place!
+  Here we override the `send()` method, copying values from the subscribed uORB topic(s) into appropriate fields in the MAVLink message, and then send the message.
+
+Next we include our new class in [mavlink_messages.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/mavlink_messages.cpp#L2193).
+Add the line below to the part of the file where all the other streams are included:
+
+```cpp
+#include "streams/CA_TRAJECTORY.hpp"
+```
+
 Finally append the stream class to the `streams_list` at the bottom of
 [mavlink_messages.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/mavlink_messages.cpp)
 
 ```C
 StreamListItem *streams_list[] = {
 ...
-create_stream_list_item<MavlinkStreamCaTrajectory>(),
+#if defined(CA_TRAJECTORY_HPP)
+    create_stream_list_item<MavlinkStreamCaTrajectory>(),
+#endif // CA_TRAJECTORY_HPP
 ...
+}
 ```
 
-Then make sure to enable the stream, for example by adding the following line to the [startup script](../concept/system_startup.md) (e.g. [/ROMFS/px4fmu_common/init.d-posix/rcS](https://github.com/PX4/PX4-Autopilot/blob/main/ROMFS/px4fmu_common/init.d-posix/rcS) on NuttX or [ROMFS/px4fmu_common/init.d-posix/rcS](https://github.com/PX4/PX4-Autopilot/blob/main/ROMFS/px4fmu_common/init.d-posix/rcS)) on SITL. Note that `-r` configures the streaming rate and `-u` identifies the MAVLink channel on UDP port 14556).
+The class is now available for streaming, but won't be streamed by default.
+We cover that in the next sections.
+
+### Streaming by Default
 
+The easiest way to stream your messages by default (as part of a build) is to add them to [mavlink_main.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/mavlink_main.cpp) in the appropriate message group.
+
+If you search in the file you'll find groups of messages defined in a switch statement:
+
+- `MAVLINK_MODE_NORMAL`: Streamed to a GCS.
+- `MAVLINK_MODE_ONBOARD`: Streamed to a companion computer on a fast link, such as Ethernet
+- `MAVLINK_MODE_ONBOARD_LOW_BANDWIDTH`: Streamed to a companion computer on slower link
+- `MAVLINK_MODE_GIMBAL`: Streamed to a gimbal
+- `MAVLINK_MODE_EXTVISION`: Streamed to an external vision system
+- `MAVLINK_MODE_EXTVISIONMIN`: Streamed to an external vision system on a slower link
+- `MAVLINK_MODE_OSD`: Streamed to an OSD, such as an FPV headset.
+- `MAVLINK_MODE_CUSTOM`: Stream nothing by default. Used when configuring streaming using MAVLink.
+- `MAVLINK_MODE_MAGIC`: Same as `MAVLINK_MODE_CUSTOM`
+- `MAVLINK_MODE_CONFIG`: ?
```

Comment:
This is used over USB by default, with a bit higher rates than NORMAL

Reply:
Thanks
```suggestion
- `MAVLINK_MODE_CONFIG`: Streaming over USB with higher rates than `MAVLINK_MODE_NORMAL`.
```